src/config.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  ChainId,
+  Config,
+  ConfigUpdate,
+  getChainById,
+  Hooks,
+  multicallAddresses,
+} from './types'
+
+function chainIdToObject<T>(val: T): Record<ChainId, T> {
+  const result: Record<number, T> = {}
+
+  const values = Object.values(ChainId)
+  values.forEach((chainId) => {
+    if (typeof chainId !== 'string') {
+      result[chainId] = val ? JSON.parse(JSON.stringify(val)) : val
+    }
+  })
+
+  return result
+}
+
+const DefaultExecutionSettings: Hooks = {
+  /* eslint-disable-next-line @typescript-eslint/no-empty-function */
+  updateCallback: () => {},
+  switchChainHook: () => Promise.resolve(undefined),
+}
+
+export const getDefaultConfig = (): Config => {
+  const defaultConfig: Config = {
+    apiUrl: 'https://li.finance/api/',
+    rpcs: chainIdToObject([]),
+    multicallAddresses: chainIdToObject(undefined),
+    defaultExecutionSettings: DefaultExecutionSettings,
+    defaultRouteOptions: {},
+  }
+
+  // RPCS
+  for (const chainIdStr of Object.keys(defaultConfig.rpcs)) {
+    const chainId = parseInt(chainIdStr) as ChainId
+    const chain = getChainById(chainId)
+    defaultConfig.rpcs[chainId] = chain.metamask.rpcUrls
+  }
+
+  // MULTICALL
+  for (const chainIdStr of Object.keys(defaultConfig.multicallAddresses)) {
+    const chainId = parseInt(chainIdStr) as ChainId
+    if (multicallAddresses[chainId]) {
+      defaultConfig.multicallAddresses[chainId] = multicallAddresses[chainId]
+    }
+  }
+
+  return defaultConfig
+}
+
+export const mergeConfig = (config: Config, configUpdate: ConfigUpdate) => {
+  // API
+  config.apiUrl = configUpdate.apiUrl || config.apiUrl
+
+  // RPCS
+  config.rpcs = Object.assign(config.rpcs, configUpdate.rpcs)
+
+  // MULTICALL
+  config.multicallAddresses = Object.assign(
+    config.multicallAddresses,
+    configUpdate.multicallAddresses
+  )
+
+  // SETTINGS
+  config.defaultExecutionSettings = Object.assign(
+    config.defaultExecutionSettings,
+    configUpdate.defaultExecutionSettings
+  )
+
+  // OPTIONS
+  config.defaultRouteOptions = Object.assign(
+    config.defaultRouteOptions,
+    configUpdate.defaultRouteOptions
+  )
```

Comment:
`Object.assign` doesn't do a deep comparison, right? So this will always overwrite the whole `defaultRouteOptions` even if the user only changed a single value inside. Do we want this behaviour or do we want a deep merge?

Reply:
It does not do a deep merge, but it should merge every top level attribute on its own and not overwrite the whole object.